### PR TITLE
Fix canceling a cursor handler via Job API

### PR DESCRIPTION
**Pull request by graetzer**
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11497/

### CODE REVIEW

File: arangod/Aql/SharedQueryState.cpp [hunk 1]
```diff
@@ -43,6 +43,7 @@ void SharedQueryState::invalidate() {
 void SharedQueryState::waitForAsyncWakeup() {
   std::unique_lock<std::mutex> guard(_mutex);
   if (!_valid) {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_QUERY_KILLED);
     return;
```
Comment (jsteemann) [inline arangod/Aql/SharedQueryState.cpp:47]:
```suggestion
```

File: scripts/startLocalCluster.sh [hunk 2]
```diff
@@ -311,5 +311,5 @@ done
 
 echo == Done, your cluster is ready at
 for p in `seq $CO_BASE $PORTTOPCO` ; do
-    echo "   ${BUILD}/bin/arangosh --server.endpoint $TRANSPORT://[::1]:$p"
+    echo "   ${BUILD}/bin/arangosh --server.endpoint '$TRANSPORT://[::1]:$p'"
```
Comment (jsteemann) [inline scripts/startLocalCluster.sh:314]: This change was not present in the 3.7 and devel versions of the PR.
Shall it be added there too?

File: arangod/Cluster/ClusterComm.h [hunk 3]
```diff
@@ -226,7 +226,7 @@ struct ClusterCommResult {
 
   /// @brief return if request was successful, use this on the request
   /// results after a `performRequest`:
-  bool successful() const {
+  bool isGood() const {
```
Comment (jsteemann) [inline arangod/Cluster/ClusterComm.h:229]: Looks like an unrelated change.
I don't understand its purpose, as the new method name `isGood` is not used/called anywhere else in this PR.
Can you clarify?

Comment (graetzer) [reply arangod/Cluster/ClusterComm.h:229]: its not used

File: scripts/startLocalCluster.sh (hunk 2 above)
Comment (graetzer) [reply scripts/startLocalCluster.sh:314]: doesn't really matter

File: arangod/Cluster/ClusterComm.h (hunk 3 above)
Comment (jsteemann) [reply arangod/Cluster/ClusterComm.h:229]: Exactly. That's why I was wondering what the purpose of the change is.
We can leave it in here, as the code doesn't exist in 3.7 anymore, so the change should not have any relevant effect for 3.6 or future versions, and doesn't need to be ported.

File: scripts/startLocalCluster.sh (hunk 2 above)
Comment (jsteemann) [reply scripts/startLocalCluster.sh:314]: Ok
